src/Pages/ADT3DScenePage/ADT3DScenePage.tsx

```diff
@@ -385,6 +395,26 @@ const ADT3DScenePageBase: React.FC<IADT3DScenePageProps> = ({
                         }
                     }
                 });
+            } else if (
+                state?.errors?.[0]?.type ===
+                    ComponentErrorType.DataFetchFailed &&
+                (state?.errors[0].rawError as any).request.status === 0 &&
```

Comment:
can we make a function to do this complex check? like `isCallSuccessful` or somehting? i see we do this in a couple spots.

Reply:
I am interested in **status code 0** specifically which happens when you cannot even make a call which is basically a "Network Error" error which might be due to CORS as well. I use this approach in other places in data history etc. So it is not just successful check but a specific status code to relate to CORS.

Reply:
ok. so then maybe we can have a getStatusCode function? It's just a lot of nested properties and null checks to get to the value we care about.